saleor/tests/fixtures.py

```diff
@@ -3972,6 +3972,20 @@ def gift_card(customer_user):
     return gift_card
 
 
+@pytest.fixture
+def gift_card2(customer_user2):
```

Comment:
That's not an obvious name for a fixture; I would not add any new ones but try to modify existing ones in the test. You can debug the resolver for that field to see why it's returning the other email (see the `resolve_created_by_email` function).

Reply:
It's happening because first it returning user.email instead of instance.created_by_email and that user.email is customer_user.email (`test@example.com`)from fixtures. 
```
return user.email if user else root.created_by_email
```
I also tried
```
    query = QUERY_GIFT_CARDS
    gift_card.user = customer_user2
    gift_card.created_by_email = customer_user2.email
    gift_card.save()
    variables = {"filter": {"createdByEmail": gift_card2.created_by_email}}
```
but it was pointing to customer_user instead of customer_user2.